### PR TITLE
Refactor unit tests

### PR DESCRIPTION
**Pull request by arturo-seijas**
Applicable spec: N/A
### Overview

Refactor unit tests

### Rationale

N/A

### Juju Events Changes

N/A

### Module Changes

N/A

### Library Changes

N/A

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

File: tests/unit/test_charm.py [hunk 1]
```diff
@@ -147,16 +171,42 @@ def test_relation_joined_when_leader_and_secrets(mock_juju_env):
...
+    act: add an smtp relation.
+    assert: the smtp relation does not populate with the password.
+    """
+    mock_juju_env.return_value = MagicMock(has_secrets=True)
+    harness = Harness(SmtpIntegratorOperatorCharm)
+    harness.set_leader(True)
+    host = "smtp.example"
+    port = 25
+    password = "somepassword"  # nosec
+    harness.update_config(
```
Comment (Thanhphan1147) [inline tests/unit/test_charm.py:190]: do you think this (the charm's config dict) can be defined in a fixture?

**Review by Thanhphan1147 — DISMISSED** [review]
LGTM, just some minor comments

**Comment by Thanhphan1147** [comment]
Also I think you forgot to fill in the PR's description

File: tests/unit/test_charm.py (hunk 1 above)
Comment (arturo-seijas) [reply tests/unit/test_charm.py]: Made a bit of refactoring

**Review by Thanhphan1147 — DISMISSED** [review]
thanks, LGTM

File: tests/unit/test_charm.py [hunk 2]
```diff
@@ -24,8 +33,8 @@ def test_unconfigured_charm_reaches_blocked_status():
 
 def test_misconfigured_port_charm_reaches_blocked_status():
     """
-    arrange: set up a charm.
-    act: trigger a configuration change with an invalid port.
+    arrange: set up a charm with an invalid port.
+    act: none
```
Comment (amandahla) [inline tests/unit/test_charm.py:37]: Why act: none?

**Review by amandahla — DISMISSED** [review]
LGTM

File: tests/unit/test_charm.py [hunk 3]
```diff
@@ -10,11 +10,20 @@
 
 from charm import SmtpIntegratorOperatorCharm
 
+MINIMAL_CHARM_CONFIG = {
+    "host": "smtp.example",
+    "port": 25,
+}
+MINIMAL_CHARM_CONFIG_WITH_PASSWORD = {
+    **MINIMAL_CHARM_CONFIG,
+    "password": "somepassword",  # nosec
```
Comment (yanksyoon) [inline tests/unit/test_charm.py:19]: I think we can use `secrets.token_hex()` here :D

**Review by yanksyoon — DISMISSED** [review]
LGTM, minor comment! Thanks!

File: tests/unit/test_charm.py (hunk 3 above)
Comment (jdkandersson) [reply tests/unit/test_charm.py:19]: yes please - that way we don't get into the habit of disabling security warnings in our code. As a side note, we do have a standard to explain why a disable is reasonable if it is required

File: tests/unit/test_charm.py (hunk 2 above)
Comment (arturo-seijas) [reply tests/unit/test_charm.py:37]: There's no action. Once the charm is set up, it reaches blocked status

File: tests/unit/test_charm.py (hunk 3 above)
Comment (arturo-seijas) [reply tests/unit/test_charm.py]: Fixed

**Review by yanksyoon — DISMISSED** [review]

**Review by yanksyoon — APPROVED** [review]

**Review by Thanhphan1147 — APPROVED** [review]

**Pull request merged**